packages/client/src/interceptors.ts

```diff
@@ -131,11 +132,43 @@ export interface WorkflowClientInterceptors {
   calls?: WorkflowClientCallsInterceptorFactory[];
 }
 
+export interface ScheduleClientCallsInterceptor {
+  /**
+   * Intercept a service call to CreateSchedule
+   */
+  create?: (input: CreateScheduleInput, next: Next<this, 'create'>) => Promise<Uint8Array /* conflictToken */>;
```

Comment:
@bergundy @lorensr 
1. Proposal specified a the return value as being the `conflictToken`, in line with `WorkflowClientCallsInterceptor.start()` and `.signalWithStart()`, which returns `runId`. I however see a risk that this might eventually cause breaking changes if we need to bubble up other fields from the grpc response. I'm considering that we should instead return the gRPC response here, or some `CreateScheduleOutput` type. Opinions?
2. Should we also intercept `update` (which can also modify headers)?
3. What about other calls? One frequent use case for interceptors is the ability to log things from there. `WorkflowClientCallsInterceptor` can intercept things like `describe` and `query`...